pkg/formats/sniffer.go

```diff
@@ -0,0 +1,96 @@
+package formats
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+)
+
+type Sniffer struct{}
+
+// SniffFile takes a path an return the format
+func (fs *Sniffer) SniffFile(path string) (Format, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return "", fmt.Errorf("opening path: %w", err)
+	}
+	return fs.SniffReader(f)
+}
+
+// SniffReader reads a stream and return the SBOM format
+func (fs *Sniffer) SniffReader(f io.ReadSeeker) (Format, error) {
+	defer f.Seek(0, 0)
+	fileScanner := bufio.NewScanner(f)
+	fileScanner.Split(bufio.ScanLines)
+
+	formatType := ""
+	formatEncoding := ""
+	formatVersion := ""
+
+	for fileScanner.Scan() {
```

Comment:
@bardenstein, I think you are interested in this type of format autodetection. It's code, so everything can be changed :) But if you wanted to weigh on any other heuristics you might want, feel free to. You can also open issues if you would prefer others.